R/f7-inputs.R

```diff
@@ -939,9 +966,16 @@ listify <- function(tag) {
     )
   ) == 1
 
-  # This list can't be styled as this would conflict
-  # with the input style.
-  if (!is_wrapped) f7List(tag) else tag
+  # This list can only be styled by inputs making use
+  # of the buildPicker function. For others, it would
+  # conflict with the input style.
+  if (!is.null(style) & !is_wrapped) {
+    do.call(f7List, c(tag, style))
+  } else if (!is_wrapped) {
+    f7List(tag)
+  } else {
+    tag
+  }
```

Comment:
@DivadNojnarg I propose here to allow the list to be styled. This can be used by `buildPicker` to style the picker, date picker and color picker. 

This allows for a more consistent look and feel with inputs that can be put in a list, and inputs that "can't":

<img width="553" alt="image" src="https://github.com/RinteRface/shinyMobile/assets/60436372/3571ac1d-b68e-4152-8864-3e138f3a79a8">

Putting the pickers directly in the list would even be better, but as I said, that is a bit tricky (that also goes for radio and checkboxes). The main issue here is the `block-title` that makes it look odd:

<img width="553" alt="image" src="https://github.com/RinteRface/shinyMobile/assets/60436372/8c27e1cf-5afd-4110-84c4-b790b303b065">

Perhaps with a bit of work we can get around that...

In any case, we would need an option to style the pickers when they are used individually, and this makes that possible. What are you thoughts?

Reply:
@hypebright: We could modify`buildPickerInput` to put the label inside the list item tag as `item-title`. This way we can also think about using `f7ListItem` in this function to reduce the code (that's a slightly modified list item with some extra classes like `item-input` ...): 

```r
buildPickerInput <- function(id, label, config, class, placeholder = NULL, style = NULL) {
  # TO DO: use f7ListItem
  inputTag <- shiny::tags$div(
    class = "item-content item-input",
    shiny::tags$div(
      class = "item-inner",
      # MOVE LABEL HERE?
      shiny::tags$div(
        class = "item-title",
        label
      ),
      shiny::tags$div(
        class = "item-input-wrap",
        shiny::tags$input(
          id = id,
          class = class,
          type = "text",
          placeholder = placeholder
        ),
        buildConfig(id, config)
      )
    )
  )

  # tag wrapper
  listify(inputTag, style)
}
```

<img width="535" alt="Screenshot 2024-04-15 at 18 36 05" src="https://github.com/RinteRface/shinyMobile/assets/18291543/ed8f049e-e111-473a-883a-6321582150bc">

Reply:
@DivadNojnarg this looks perfect! If we can have this and keep the original look outside a list it’s a winner 🏆